docs/tutorials.rst

```diff
@@ -250,41 +258,39 @@ your application.
 .. code-block:: python
    :linenos:
 
-    import random
-    import string
-
-    import cherrypy
-
-    class StringGenerator(object):
+   import random
+   import string
+   
+   import cherrypy
+   
+   
+   class StringGenerator(object):
        @cherrypy.expose
        def index(self):
            return """<html>
              <head></head>
-         <body>
-           <form method="get" action="generate">
-             <input type="text" value="8" name="length" />
+             <body>
+               <form method="get" action="generate">
+                 <input type="text" value="8" name="length" />
                  <button type="submit">Give it now!</button>
-           </form>
-         </body>
-       </html>"""
-
+               </form>
+             </body>
+           </html>"""
+   
        @cherrypy.expose
        def generate(self, length=8):
            some_string = ''.join(random.sample(string.hexdigits, int(length)))
            cherrypy.session['mystring'] = some_string
            return some_string
-
+   
        @cherrypy.expose
        def display(self):
            return cherrypy.session['mystring']
-
-    if __name__ == '__main__':
-        conf = {
-            '/': {
-                'tools.sessions.on': True
-            }
-        }
-        cherrypy.quickstart(StringGenerator(), '/', conf)
+   
+   
+   if __name__ == '__main__':
+       conf = {'/': {'tools.sessions.on': True}}
```

Comment:
Thanks for introducing me to yapf. I was not familiar with it, and may consider using it in my work. Regarding this issue in particular, though, the FAQ has a section on [Why Does YAPF Destroy My Awesome Formatting](https://github.com/google/yapf#why-does-yapf-destroy-my-awesome-formatting). Their workaround is to add comments to the formatter to ignore these sections. Unfortunately, I don't think including these comments is a good idea here, in the tutorial setting.